text/0000-location.md

```diff
@@ -0,0 +1,409 @@
+Location entity as well as a more specific - and fully encapsulated - GS1 
+compliant Location. GS1 is a widely used data standard in enterprises and, given 
+that positioning and familiarity, Grid support feels natural. Additional 
+implementations of Location for specialized industries or use cases may derive 
+from or extend this implementation.
+
+This Grid Location proposal is specific to physical locations within a supply 
+chain. The GS1 General Specification defines a physical location as a site (an area,
+a structure or group of structures) or an area within the site where something 
+was, is, or will be located. For example: A label attached to a loading dock or 
+to a shelf location in a warehouse. Legal entities, functions and digital 
+locations are out of scope for our initial design and should be addressed in 
+supplemental RFC(s).
+
+# Motivation
+[motivation]: #motivation
+
+The Grid Location implementation is designed for sharing location master data 
+between participants. Location is a near universal concept within supply chain 
+solutions and would naturally be one of the highest areas of re-use across Grid 
+applications. The identification of where a physical location exists (and thus 
+where a business transaction occurs) is a foundational element that will enable 
+a wide host of distributed supply chain and commerce solutions. The design will 
+address use cases such as:
+
+- the inclusion of Location within business transactions (e.g. track and trace 
+events, inventory and warehouse management, point of sale, network optimization).
+- the enrichment of UX experiences by the inclusion of additional Location 
+attribution (e.g. name, address, location type).
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+[entities]: #entities
+
+A location has an identifier, a namespace, an owner, and one or more properties. 
+Properties are described in the �_Defined GS1 Properties_� section of the RFC.
+
+|Master Data Common Name|Attribute Name|Description|Example|
+|-----------------------|--------------|-----------|-------|
+|Location Identifier|location_id|A location is referenced using a location_id (key attribute). For GS1 locations, the location_id is a Global Location Number which is part of GS1 specifications.|"0099474000005"|
+|Namespace|location_namespace|This RFC defines a single location_namespace for GS1; a location in the GS1 location_namespace is called a GS1 location. Note that this design supports extending additional location namespaces in the future.|"01"|
+|Owner|org_id|The identifier of the organization responsible for maintaining the location. For GS1 locations, org_id is the Pike organization which claims ownership of the GS1 company prefix (via the "gs1_company_prefixes" field in the Pike records). The GLN will always start with the company prefix. The smart contract will check that the org claiming ownership lists the matching prefix in their Pike org.||
```

Comment:
GS1 US standards team member here...

Great to see the connection between the company that is licensing a GS1 company prefix (GCP) and the location IDs they would be issuing. It's important to note that some GS1 Member Organizations license individual GS1 Identification Keys instead of a GCP. In those instances, attributing a GCP to the 'Owner' in Grid would not be accurrate.

Reply:
@visibleOrigins in this case, is the prefix actually held by the GS1 Member Organization? Would the "owner" of the key in that case be GS1 who would be issuing the record on behalf of the licensing party?

Reply:
Using the definition "The identifier of the organization responsible for maintaining the location.", it's still accurate to point to the organization which has licensed the GLN not the licensing GS1 MO. 

In the same vein, it would be better to say that the "...org_id is the Pike organization which claims _a license to_ the GS1 company prefix..."

Reply:
@visibleOrigins we ultimately envision a model where we have a strong trust linkage between the domain identity (the GS1 Company Prefix) and the ledger identity (the organizational/key schemes which allow authorized agents to sign transactions). For example, we could link the "party responsible for maintaining identifiers under this GS1 prefix" to a given Pike organization (maybe with a transaction signed by the trusted GS1 identity...). This would allow us to gate the authenticity of record updates if and only if they are signed by the trusted organization. Cargill, managing the `0653114` prefix would be able to publish GTINs, GLNs, etc. starting with that prefix and signed by a key authorized by the linked Cargill organization. A hypothetical 'Bad Co', not having that chain of trust, would not be able to publish valid records under the `0653114` prefix.

Can we extend this model when single GLNs are being licensed out of a centrally managed prefix? It would seem to break down to me - unless the manager of the centrally managed prefix is the one authoritatively publishing the records. In this case, the MO would be provided the GLN attributes by the licensor and the MO would be the publisher of the record, presumably after some side-band vetting process to make sure the licensor is providing quality data.

Another possibility would be declaring ranges of the company prefix namespace as "unenforced" where we would just allow any identity to publish any record in that range.

Reply:
Thanks @jsmitchell.  Awesome to see this RFC being created.  Would there be service or contractual cost involved with getting a transaction signed by the trusted GS1 identity as you mention above? How would we get a transaction signed by a trusted GS1 identity?

Reply:
@MattP007 I can't speak for GS1 and any sort of operational/publishing signed claims role in the future. I was speaking more toward a vision of what we could build and how it would interact with the existing prefix design in GS1. Rationalizing record ownership and tying it to authoritative parties via a chain of trust seems like an important capability to me.

Reply:
Thanks @jsmitchell.  I agree that rationalizing record ownership and tying it to authoritative parties via a chain of trust is an important capability.  What I'm trying to understand is if we are signing up for a cost here so we are going in with a common understanding.

Reply:
Last I looked into this I thought this was a paid service from GS1 as part of their GS1 US data hub service, but it's been so long maybe I'm not recalling that correctly.  If that's the case, that may be fine, I just want to understand that more.  If I'm not recalling that correctly, I'll have to go back and review my notes and understanding from last time I spoke with folks at GS1. Or maybe one of the GS1 folks on the thread can clarify.

Reply:
I am trying and not succeeding in understanding what specific interaction you are indicating needs to occur to link a Pike ID to a GS1 location, the owner of which is that entity owning the GCP (which btw has a GLN associated wit the owning company of the GCP) activity.  GS1 does have a service - Data Hub Location - that when given a GLN - if it finds a match, it provides all of the attribute data for that GLN (including the owning entity of the GCP for which that GLN is created from with and exception being group purchasing orgs (they may own the GCP but the GLNs belong to members OR in the instance of single GLN). 

GS1 Data Hub Location is a subscription. View/Use is $500. 

But - as I said, the verification you are looking for is not entirely clear.

Reply:
I'm assuming even when licensing individual keys that GS1 assigns a unique company prefex for each licensing org (and that prefix is used across individually licensed keys).  

Could MO have a limited GPC context?  (For example, would a license only allow GTINs to be created but not GLN?)

Reply:
@genasmorgan Thanks for that info.  Does the $500 include access to the developer API or is there a separate cost for API access?

Reply:
Let's please move the GS1 Data Hub costs/access conversation somewhere else.  Want to keep this conversation focused on the design of Grid Location.

Reply:
@davececchi Happy to move the conversation elsewhere.  I think a better understanding of the design intent and related costs to the design is needed before we get more involved.  I agree with @genasmorgan that the verification you are looking for is not entirely clear.

Reply:
Overall design intent is explained above in the Summary and Motivation sections.  James' comments speak to the specific design nuance we are speaking to here.

Thanks for taking the rest offline so we can get back to the discussion.

Reply:
To clarify - for entities who license ONE GLN, these are truly single-issued GLNs; meaning the GS1 Member Organization may assign many GLNs out of what is traditionally known as a "prefix" to individual, non-related organizations. @davececchi, to your question about limited GPC context, we do not have that. If you license a prefix, you can use it to create ANY GS1 key (GTIN, GLN, GIAI, GSRN, etc.). You are only limited by the capacity you license. In other words, each GS1 key is fixed length, and your whole prefix would be included in any key, with the remainder of the digits in the key left for you to assign accordingly. For example, if you license a 6-digit prefix from GS1 US, you can create up to 100,000 unique GTINs or GLNs, but if you license a 10-digit prefix from GS1 US, you can only create up to 10 unique GTINs or GLNs. The main thing is that a GS1 Member Organization would never assign a single GLN such that it would overlap a prefix licensed to another organization. Uniqueness of the keys and capacity we license is the critical foundation.